vnext/Shared/Networking/WinRTHttpResource.cpp

```diff
@@ -205,46 +154,113 @@ void WinRTHttpResource::UntrackResponse(int64_t requestId) noexcept {
   m_responses.erase(requestId);
 }
 
-fire_and_forget
-WinRTHttpResource::PerformSendRequest(int64_t requestId, HttpRequestMessage &&request, bool textResponse) noexcept {
+fire_and_forget WinRTHttpResource::PerformSendRequest(HttpRequestMessage &&request, IInspectable const &args) noexcept {
   // Keep references after coroutine suspension.
   auto self = shared_from_this();
   auto coRequest = std::move(request);
+  auto coArgs = args;
+  auto coReqArgs = winrt::get_self<RequestArgs, IInspectable>(coArgs);
 
   // Ensure background thread
   co_await winrt::resume_background();
 
+  HttpMediaTypeHeaderValue contentType{nullptr};
+  string contentEncoding;
+  string contentLength;
+
+  // Headers are generally case-insensitive
+  // https://www.ietf.org/rfc/rfc2616.txt section 4.2
+  for (auto &header : coReqArgs->Headers) {
+    if (_stricmp(header.first.c_str(), "content-type") == 0) {
+      bool success = HttpMediaTypeHeaderValue::TryParse(to_hstring(header.second), contentType);
+      if (!success && m_onError) {
+        co_return m_onError(coReqArgs->RequestId, "Failed to parse Content-Type");
+      }
+    } else if (_stricmp(header.first.c_str(), "content-encoding") == 0) {
+      contentEncoding = header.second;
+    } else if (_stricmp(header.first.c_str(), "content-length") == 0) {
+      contentLength = header.second;
+    } else if (_stricmp(header.first.c_str(), "authorization") == 0) {
+      bool success =
+          coRequest.Headers().TryAppendWithoutValidation(to_hstring(header.first), to_hstring(header.second));
+      if (!success && m_onError) {
+        co_return m_onError(coReqArgs->RequestId, "Failed to append Authorization");
+      }
+    } else {
+      coRequest.Headers().Append(to_hstring(header.first), to_hstring(header.second));
+    }
+  }
+
+  IHttpContent content{nullptr};
+  if (BodyData::Type::String == coReqArgs->Body.Type) {
```

Comment:
>coReqArgs->Body.Type [](http://example.com/codeflow?start=32&length=20)

nit, should we locally cache this expression? 
` auto bodyType = coReqArgs->Body.Type`

Reply:
Can you elaborate?
The "type" is just an enum (int).

If you mean "caching" as in keeping a reference for coroutine usage, that's solved by declaring `coReqArgs`.

Reply:
yes I meant local var... to store the value i.e. 
```cpp
auto bodyType = coReqArgs->Body.Type;
if (BodyData::Type::String == bodyType) {
 ....
```

Reply:
note: not that important...

Reply:
Huh, maybe. Being just a struct member, I don't think it has a performance implication. Maybe the compiler optimizes its access?